hooks/tk-multi-publish2/basic/publish_session.py

```diff
@@ -0,0 +1,712 @@
+
+        self.logger.info(
+            "Publish created for file: %s" % (publish_path,),
+            extra={
+                "action_show_in_shotgun": {
+                    "label": "Show Publish",
+                    "tooltip": "Open the Publish in Shotgun.",
+                    "entity": publish_data
+                }
+            }
+        )
+
+        # insert the path into the properties
+        item.properties["next_version_path"] = self._bump_file_version(
+            path, settings)
+
+    def _bump_file_version(self, path, settings):
+        """
+        Save the supplied path to the next version on disk.
+        """
+
+        (next_version_path, version) = self._get_next_version_info(
+            path, settings)
+
+        if version is None:
+            self.logger.debug(
+                "No version number detected in the publish path. "
+                "Skipping the bump file version step."
+            )
+            return None
+
+        self.logger.info("Incrementing session file version number...")
+
+        # nothing to do if the next version path can't be determined or if it
+        # already exists.
+        if not next_version_path:
+            self.logger.warning("Could not determine the next version path.")
+            return None
+        elif os.path.exists(next_version_path):
+            self.logger.warning(
+                "The next version of the path already exists",
+                extra={
+                    "action_show_folder": {
+                        "path": next_version_path
+                    }
+                }
+            )
+            return None
+
+        # save the session to the new path
+        _save_session(next_version_path)
+        self.logger.info("Session saved as: %s" % (next_version_path,))
+
+        return next_version_path
+
+    def _get_next_version_info(self, path, settings):
+        """
+        Return the next version of the supplied path.
+
+        If templates are configured, use template logic. Otherwise, fall back to
+        the zero configuration, path_info hook logic.
+
+        :param str path: A path with a version number.
+        :param settings: Configured settings for this plugin.
+
+        :return: A tuple of the form::
+
+            # the first item is the supplied path with the version bumped by 1
+            # the second item is the new version number
+            (next_version_path, version)
+        """
+
+        publisher = self.parent
+
+        # get the configured work file template
+        work_template = self._get_template("Work file Template", settings)
+
+        # set these so we can check to see if the template logic applied
+        next_version_path = None
+        version = None
+
+        if work_template:
+            # if the work file template matches, we'll get parsed fields
+            fields = work_template.validate_and_get_fields(path)
+
+            if fields and "version" in fields:
```

Comment:
Use template-based logic if possible.

Reply:
debug logging here would be great. maybe do a general pass too and see if it makes sense in other places?

Reply:
will do.